test.l

```diff
@@ -1,31 +1,34 @@
 #!/usr/bin/env bin/lumen
 
+(when-compiling
+  (define-macro test (x msg)
+    `(if (not ,x)
+         (do (set failed (+ failed 1))
+             (return ,msg))
+       (inc passed)))
+
+  (define-macro test= (a b)
+    (let-unique (x y)
+      `(let (,x ,a ,y ,b)
+         (test (equal? ,x ,y)
+               (cat "failed: expected " (str ,x) ", was " (str ,y))))))
+
+  (define-macro define-test (name rest: body)
+    `(add tests (list ',name (fn () ,@body))))
+
+  nil)
```

Comment:
To clarify, without that `nil`, you'd get an error:

```
> (when-compiling
    (define-macro add1 (x)
      `(+ ,x 1))
    nil)
> (add1 2)
3
> (when-compiling
    (define-macro add1 (x)
      `(+ ,x 1)))
error: attempt to index local 'l' (a function value)
stack traceback:
   ...
```

The reason for the error is because the output of the above `define-macro` statement is `(setenv 'add1 macro: (fn (x) (+ x 1)))`. So that would be equivalent to this:

```
> (when-compiling
    (setenv 'add1 macro: (fn (x) (+ x 1))))
error: attempt to index local 'l' (a function value)
```

and since the result of calling `setenv` is returned at compile time, it causes an error.